android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceIdResolver.java

```diff
@@ -0,0 +1,54 @@
+package com.learnium.RNDeviceInfo.resolver;
+
+import android.content.Context;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * Instance Id resolver's single purpose is to get the device's Instance Id
+ * author: Andres Aguilar
+ */
+public class DeviceIdResolver {
+
+    private final Context context;
+
+    public DeviceIdResolver(Context context){
+        this.context = context;
+    }
+
+    public String getInstanceIdSync() {
+        try {
+            return getFirebaseInstanceId();
+        } catch (ClassNotFoundException e) {
+            System.err.println("N/A: Add com.google.firebase:firebase-iid to your project.");
```

Comment:
The only thing I don't like about this is that structurally library consumers will definitely get a System.err.println() with a dire message, but then just after getGmsInstanceId() might succeed meaning things are deprecated but not technically in error.

Is there a way to more gently handle the failure in getInstanceIdSync - ideally with the catches setting flag so you can then offer different messaging if both are missing vs one, but recognizing that is a larger restructure, even a slight textual change acknowledging GMS still exists but firebase is the future vs text that sounds like a hard-fail would work for me

Reply:
Hello, I modified so it only writes a message if both classes are not found. Thank you